BookstoreApp/Readme.md

```diff
@@ -0,0 +1,120 @@
+Response: No content with status 204 if successful, 404 if the book with the given ID is not found.
+- GET: /books/searchByTitle
+Search books by title.
+Request Parameter: Title of the book to search for.
+Response: List of books with status 200.
+- GET: /books/searchByAuthor
+Search books by author.
+Request Parameters: First name and last name of the author.
+Response: List of books with status 200.
+- GET: /books/searchByGenre
+Search books by genre.
+Request Parameter: Genre of the book.
+Response: List of books with status 200.
+### Author Controller:
+
+- POST: /authors
+Create a new author.
+Request Body: Author object.
+Response: Created author with status 201.
+GET: /authors
+- Get a list of all authors.
+Response: List of authors with status 200.
+- GET: /authors/{id}
+Get an author by its unique ID.
+Path Variable: ID of the author.
+Response: Author with status 200 if found, 404 if not found.
+- PUT: /authors/{id}
+Update an author by its ID.
+Path Variable: ID of the author to be updated.
+Request Body: Updated author object.
+Response: Updated author with status 200 if successful, 404 if the author with the given ID is not found.
+- DELETE: /authors/{id}
+Delete an author by its ID.
+Path Variable: ID of the author to be deleted.
+Response: No content with status 204 if successful, 404 if the author with the given ID is not found.
+
+### Genre Controller:
+
+- POST: /genre
+Create a new genre.
+Request Body: Genre object.
+Response: Created genre with status 201.
+- GET: /genre
+Get a list of all genres.
+Response: List of genres with status 200.
+- GET: /genre/{id}
+Get a genre by its unique ID.
+Path Variable: ID of the genre.
+Response: Genre with status 200 if found, 404 if not found.
+- PUT: /genre/{id}
+Update a genre by its ID.
+Path Variable: ID of the genre to be updated.
+Request Body: Updated genre object.
+Response: Updated genre with status 200 if successful, 404 if the genre with the given ID is not found.
+- DELETE: /genre/{id}
+Delete a genre by its ID.
+Path Variable: ID of the genre to be deleted.
+Response: No content with status 204 if successful, 404 if the genre with the given ID is not found.
+
+
+## Code quality
+
+| Parameter | result |
+|:---------:| :---: |
+|test coverage (service and controller classes) | 100% |
+
+
+### questions from the task
+
+- Was it easy to complete the task using AI? _**yes**_
+- How long did task take you to complete? (Please be honest, we need it to gather anonymized statistics) _**5 hours**_
+- Was the code ready to run after generation? _**mostly**_
+- What did you have to change to make it usable? _**correct some small mistakes, configuration**_
+- Which challenges did you face during completion of the task? _**spring configuration(different version compatibility)**_
+- Which specific prompts you learned as a good practice to complete the task? _**any specific, I tried different**_
+
+
+chatGPT logs:
+https://chat.openai.com/share/531beeb6-a6bd-4dc9-b46d-d9cdc260f1e9
```

Comment:
This link is for first task.

Reply:
sorry, right link is: https://chat.openai.com/share/af7e436c-b8b1-4bfc-b4c2-b91038b07516